packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -0,0 +1,372 @@
+// Copyright 2019 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/gestures.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'layout_builder.dart';
+import 'scroll_context.dart';
+import 'scroll_controller.dart';
+import 'scroll_physics.dart';
+import 'scroll_position.dart';
+import 'scroll_position_with_single_context.dart';
+import 'scroll_simulation.dart';
+
+/// The signature of a method that provides a [BuildContext] and
+/// [ScrollController] for building a widget that may overflow the draggable
+/// [Axis] of the containing [DraggableScrollSheet].
+///
+/// Users should apply the [scrollController] to a [ScrollView] subclass, such
+/// as a [SingleChildScrollView], [ListView] or [GridView], to have the whole
+/// sheet be draggable.
+typedef ScrollableWidgetBuilder = Widget Function(
+  BuildContext context,
+  ScrollController scrollController,
+);
+
+/// A widget that can be dragged along the vertical axis between its
+/// minChildSize, which defaults to `0.25` and maxChildSize, which defaults
+/// to `1.0`. These sizes are percentages of the height of the parent container.
```

Comment:
The first paragraph should just be one line and it should summarize what the widget does. Here's a suggestion:

A container for a [Scrollable] that responds to drag gestures by resizing the scrollable until a limit is reached, and then scrolling.